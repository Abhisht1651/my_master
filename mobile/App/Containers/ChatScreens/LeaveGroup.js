// @flow
import React from 'react'
import { View, Text, TouchableOpacity, Modal, Image } from 'react-native'
import Images from '../../Themes/Images'
import { MediumThinButton } from '../../Components/Buttons/MediumThinButton'

// Styles
import styles from './ScreenStyles/LeaveGroupStyles'

type Props = {
  isModalVisible: boolean,
  onBtnPress: () => void,
  onClose: () => void
}

const LeaveGroup = (props: Props) => {
  const { isModalVisible, onBtnPress, onClose } = props
  return (
    <Modal visible={isModalVisible} transparent>
      <View style={styles.screen}>
        <View style={styles.innerScreen}>
          <View style={styles.view1}>
            <TouchableOpacity style={styles.closeIconContainer} activeOpacity={1} onPress={onClose}>
              <Image source={Images.closeIcon} style={styles.closeIcon} />
            </TouchableOpacity>
          </View>
          <View style={styles.view2}>
            <Text style={styles.headerText}>Do you want to leave</Text>
            <Text style={styles.headerText}>the group?</Text>
            <View style={styles.verticalgap} />
            <Text style={styles.headerText2}>You will not be able to see new</Text>
            <Text style={styles.headerText2}>notifications from this chat.</Text>
          </View>
          <View style={styles.view3}>
            <MediumThinButton title="Leave" onPress={onBtnPress} />
          </View>
        </View>
      </View>
    </Modal>
  )
}
LeaveGroup.defaultProps = {
  isModalVisible: true
}

export default LeaveGroup
