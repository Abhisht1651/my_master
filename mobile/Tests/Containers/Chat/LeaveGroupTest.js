import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, Image } from 'react-native'
import LeaveGroup from '../../../App/Containers/ChatScreens/LeaveGroup'

describe('LeaveGroup', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<LeaveGroup />) // no compile-time error
  })
  it('LeaveGroup component renders correctly', () => {
    const wrapper = shallow(<LeaveGroup />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 4 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(4)
  })
  it('should render 1 Image', () => {
    expect(wrapper.find(Image)).toHaveLength(1)
  })
})
