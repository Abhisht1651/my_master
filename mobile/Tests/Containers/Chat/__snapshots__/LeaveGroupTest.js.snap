// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveGroup LeaveGroup component renders correctly 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2222234D",
        "height": 1334,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "height": 527.3571428571429,
          "width": 650,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#E6ECED",
            "flexDirection": "row",
            "height": 87.07142857142857,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ForwardRef
          activeOpacity={1}
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": 41.94,
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "aspectRatio": 1,
                "height": 17.28,
              }
            }
          />
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 30,
              "letterSpacing": -0.14,
            }
          }
        >
          Do you want to leave
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 30,
              "letterSpacing": -0.14,
            }
          }
        >
          the group?
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "height": 26.285714285714285,
              "width": "100%",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#262676",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 24,
              "letterSpacing": -0.11,
            }
          }
        >
          You will not be able to see new
        </Text>
        <Text
          style={
            Object {
              "color": "#262676",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 24,
              "letterSpacing": -0.11,
            }
          }
        >
          notifications from this chat.
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 65.71428571428571,
            "marginTop": "auto",
          }
        }
      >
        <MediumThinButton
          disabled={false}
          onPress={[Function]}
          style={Object {}}
          title="Leave"
        />
      </View>
    </View>
  </View>
</Component>
`;
